galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/commons/QueryParameters.java

```diff
@@ -127,6 +128,34 @@ public Instant getSingleInstant(String queryParameterName, Instant defaultValue)
 		return returnedValue;
 	}
 
-	
-
+	public Instant getSingleInstantIfParameterNotPresent(String queryParameterName , String parameterToTest) throws InternalServletException {
+		// This function will return null when neither of the two parameters provided are in the query
+		String paramTestValueStr = getSingleString(parameterToTest, null);
+		Instant result = getSingleInstant(queryParameterName, null);
+		if (paramTestValueStr == null && result ==null) {
+			return null;
+		}
+		return result;
+	}
+		
+	public Instant getDefaultFromInstantIfNoQueryIsPresent () throws InternalServletException{
+		// The default for 'from' is now-24 hours. If no query parameters are specified
+		Integer querysize = this.params.size();
+		Instant from = null ;
+		if (querysize >= 0){
+			from = getSingleInstantIfParameterNotPresent("from", "runname");
+			//Check to see if there is no query (i.e. hit the /ras/runs/ endpoint)
+			if (from == null && querysize == 0){
+				Instant fromDefault = Instant.now().minus(24, ChronoUnit.HOURS);
+				from = getSingleInstant("from", fromDefault);
+			}else if (from == null){
+				/*  RULE: Throw exception because a query exists but no from date has been supplied
```

Comment:
Use // comments here please... I tend to not want /* ... */ in the code as it stops me commenting-out blocks and methods when I want to see what the dependencies are... sometimes.